### PR TITLE
Throttle inflight activations when exceeding max count

### PR DESCRIPTION
**Pull request by enochtangg**
This PR is responsible for making the following changes:
* Move `upkeep` step to the beginning of the processing pipeline.
* Add `limit` step after do `upkeep` which is responsible for polling inflight activation store and ensure that the pending count does not exceed limit set. If it does, wait until workers finishing processing before writing new batch to store.

Testing:
Set `self.max_inflight_activation_in_store=1` and produce multiple tasks consecutively. The consumer will wait until worker has completed its single task before writing to the store again.

### CODE REVIEW

File: src/sentry/taskworker/processors/strategy_factory.py [hunk 1]
```diff
@@ -86,18 +90,32 @@ def flush_batch(
...
+            count = self.pending_task_store.count_pending_task()
+            while count >= self.max_inflight_activation_in_store:
+                # The number of pending inflight activations in the store exceeds the limit.
+                # Wait for workers to complete tasks before adding the next offset to the queue.
+                logger.info(
+                    "Number of inflight activation: %s exceeds the limit: %s. Retrying in 3 seconds",
+                    count,
+                    self.max_inflight_activation_in_store,
+                )
+                time.sleep(3)  # TODO: make this a parameter
```
Comment (john-z-yang) [inline src/sentry/taskworker/processors/strategy_factory.py:113]: We should emit backpressure by raising a `MessageRejected` error.

It's not recommended to sleep in submit.
https://getsentry.github.io/arroyo/backpressure.html#backpressure

[the remark this answers is not in the document]
Comment (john-z-yang) [reply src/sentry/taskworker/processors/strategy_factory.py:113]: We can just raise `MessageRejected`

Comment (enochtangg) [reply src/sentry/taskworker/processors/strategy_factory.py]: Good callout, and kudos to whoever wrote the documentation for it. Let me know how the changes look.

**Review by markstory — APPROVED** [review]
Looking good :clap:

**Pull request merged**